CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeEventManager.cs

```diff
@@ -170,6 +181,7 @@ private void SwitchOrCreateProfile(Dictionary<string, object> profile, string ca
                     _coreState.DeviceInfo.ForceNewDeviceID();
                 }
 
+                SetResponseInterceptors();
```

Comment:
Why this needs to be set again here?

Reply:
To update the ResponseInterceptors with new DeviceId.